tests/testthat/test-Model-methods.R

```diff
@@ -21,6 +21,30 @@ test_that("doseFunction-GeneralModel returns correct dose function", {
   expect_identical(dose_fun_env[["samples"]], samples)
 })
 
+test_that("doseFunction-GeneralModel returns correct dose function for matrix param", {
+  model <- h_get_logistic_log_normal_mix()
+  params <- list(
+    alpha0 = matrix(c(-0.94, -0.94, -2.37, -2.37, -0.67, -0.67, -1.28, -1.08), nrow = 4),
+    alpha1 = matrix(c(0.45, 0.45, 0.40, 0.40, 0.75, 0.75, 1.18, 0.63), nrow = 4),
+    comp = c(1, 1, 1, 1)
+  )
+  samples <- h_as_samples(params, burnin = 10000, fixed = FALSE)
+  dose_args <- c("x", "model", "samples")
+
+  dose_fun <- doseFunction(model, alpha0 = params$alpha0, alpha1 = params$alpha1, comp = params$comp)
+  dose_fun_dose_args <- as.character(body(dose_fun)[[2]][-1])
```

Comment:
this is too complicated for me - what is this? please add comment and check if cannot be simplified

Reply:
This is:
```
> dose_fun
function(x) {
      dose(x, model, samples)
    }
<environment: 0x7fd861415190>
> dose_fun_dose_args <- as.character(body(dose_fun)[[2]][-1])
> dose_fun_dose_args
[1] "x"       "model"   "samples"
```
so with `dose_fun_dose_args <- as.character(body(dose_fun)[[2]][-1])` I am trying to get the argument names from the inner dose function.

Reply:
Aha yeah please add comments

Reply:
Hi @danielinteractive I slightly improved the code (add more robust and flexible checks) and also added comments. I did it only for `doseFunction-GeneralModel returns correct dose function` test. Please let me know if this looks ok so that I will propagate it to other tests.